doc/StructuredTypes.md

```diff
@@ -0,0 +1,187 @@
+## Concept
+
+Snowflake structured types documentation is available here: [Snowflake Structured Types Documentation](https://docs.snowflake.com/en/sql-reference/data-types-structured).
+
+Snowflake offers a way to store structured types which can be:
+- objects, e.g. ```OBJECT(city VARCHAR, state VARCHAR)```
+- arrays, e.g. ```ARRAY(NUMBER)```
+- maps, e.g. ```MAP(VARCHAR, VARCHAR)```
+
+The driver allows reading and casting such structured objects into customer classes.
+
+**Note**: Currently, reading structured types is available only for JSON result format.
+
+## Enabling the feature
+
+Reading structured types is available only for JSON result format, so you can make sure you are using JSON result format by:
+```sql
+ALTER SESSION SET DOTNET_QUERY_RESULT_FORMAT = JSON;
+```
+
+You can enable the feature by setting session parameters:
+```sql
+ALTER SESSION SET ENABLE_STRUCTURED_TYPES_IN_CLIENT_RESPONSE = true;
+ALTER SESSION SET IGNORE_CLIENT_VESRION_IN_STRUCTURED_TYPES_RESPONSE = true;
```

Comment:
veSRion

Reply:
done

Reply:
In my opinion the parameter `IGNORE_CLIENT_VESRION_IN_STRUCTURED_TYPES_RESPONSE` isn't public and should be used only for development. In the end you need to add the .NET version to https://github.com/snowflakedb/snowflake/blob/26243dfe80a0eec23b2ec110992e07bde3c663d7/GlobalServices/src/main/java/com/snowflake/core/FeatureWithClientSupport.java#L213C3-L213C27